packages/ds-ext/src/DiagramEditorProvider.ts

```diff
@@ -38,15 +39,28 @@ export class DiagramEditorProvider implements vscode.CustomEditorProvider<Diagra
   }
 
   dispose(): void {
-    this.duckDBStorage.close();
+    if (this.duckDBStorage instanceof DuckDBStorage) {
+      this.duckDBStorage.close();
+    }
   }
 
-  constructor(private readonly context: vscode.ExtensionContext) {
-    const dbPath = path.join(vscode.workspace.workspaceFolders![0].uri.fsPath, 'diagram.db');
-    this.duckDBStorage = new DuckDBStorage(dbPath);
+  private async init(): Promise<void> {
+    const dbPath = createDataStoryDBPath();
```

Comment:
Similar to other comments, this exposes implementation details of the DuckDb solution. Could this code as well be hidden behind the interface?